File: Asgard_Test/tests_Heimdall/benchmarks/corpus/taint_js/tp_sa4_nonvalidator_guard_still_flags.js
const { exec } = require('child_process');

// SA4 path-sensitivity (mute-bug guard case): `host !== null` is NOT a
// catalog-verified value-domain predicate -- it proves nothing about the
// character content of `host`. The engine must NOT invent a semantic
// validator here; the guarded path must still flag.
app.post('/ping', (req, res) => {
    const host = req.body.host;
    if (host !== null) {
        exec('ping -c 1 ' + host, (err, stdout) => {
            res.send(stdout);
        });
    }
});

File: Asgard_Test/tests_Heimdall/benchmarks/corpus/taint/tp_sa4_nonvalidator_guard_still_flags.py
"""TP: SA4 mute-bug guard case -- `host is not None` is NOT a catalog-
verified value-domain predicate; it proves nothing about the character
content of `host`. The engine must NOT invent a semantic validator here --
the guarded path must still flag."""

import subprocess


def run():
    host = request.args.get("host")
    if host is not None:
        subprocess.run("ping -c 1 " + host, shell=True)
